docs/conf.py

```diff
@@ -295,6 +295,12 @@ def _custom_edit_url(
         if module := getattr(item, "__module__", None):
             mod_dir, _, mod_file = module.rpartition(".")
             new_file_name = mod_file + ".py"
+            # Remove wrappings, such as functools.cache in astropy.units.equivalencies.
+            # This also avoids "could not find source" warnings.  But don't remove
+            # a correct one for sharedmethod.
+            if module != "astropy.utils.decorators":
+                while wrapped := getattr(item, "__wrapped__", None):
```

Comment:
I cannot grok if this will have side effects or not... I'll just trust you on this one, as long as the warnings are gone from the logs. That "edit" button was never that useful anyway.

Reply:
With this change, the `edit` button is actually back! And it points to the right place.